app/src/main/java/com/epam/havina/task1_recycler/RecyclerAdapter.java

```diff
@@ -0,0 +1,58 @@
+package com.epam.havina.task1_recycler;
+
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+final class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.ItemHolder> {
+
+    private final int[] itemsList;
+    private final ItemClickListener listener;
+
+    RecyclerAdapter(@NonNull final ItemClickListener listener,
+                    @NonNull final int[] itemsList) {
+        this.listener = listener;
+        this.itemsList = itemsList;
+    }
+
+    @NonNull
+    @Override
+    public ItemHolder onCreateViewHolder(@NonNull final ViewGroup parent, final int viewType) {
+        return new ItemHolder(LayoutInflater
```

Comment:
I would like you to suggest this solution: 

`final View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.recycler_view_item, parent, false);

        final RecyclerView.ViewHolder holder = new ItemHolder(view);
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                final int adapterPosition = holder.getAdapterPosition();
                if (adapterPosition != RecyclerView.NO_POSITION) {
                    listener.onItemClicked(itemsList[adapterPosition]);
                }
            }
        });

        return (ItemHolder) holder;`